framework/src/base/MooseApp.C

```diff
@@ -1579,6 +1579,8 @@ MooseApp::getCheckpointDirectories() const
 
   // Add the directories added with Outputs/checkpoint=true input syntax
   checkpoint_dirs.push_back(getOutputFileBase() + "_cp");
+  // Add the directories added with the autosave checkpoint input syntax
+  checkpoint_dirs.push_back(_output_file_base + "_autosave_cp");
```

Comment:
I don't think you need this line anymore.

Reply:
So, this is still here (was actually readded in the last stages of development) because its the simplest solution I found to the directory naming problems.

Basically, whatever you pass in for the second param in `_problem->addOutput("Checkpoint", "autosave", cp_params)` gets appended to the output directory name. You can't pass an empty string into it either. This means the options for collecting the checkpoint directories of the autosave are to either:

- Do a bunch of ugly string manipulation when outputting the autosave checkpoint to not contain `autosave_cp` in the directory name
- Create it equivalently to the regular checkpoints, which runs a sizable amount of operations to determine the output path
- Simply import "autosave_cp" directories when running a recover.

I chose the last option because its the least time both performance-wise and dev-time-wise. 

It's important to note that it isn't possible for there to be autosave outputs and regular checkpoint outputs on the same test -- the autosave checkpoint only gets created if there _isn't_ an existing checkpoint object in the environment, and so there will never be overlapping files or concerns about selecting the most recent checkpoint output. This is actually the best solution to the problem.

Reply:
Ok. We may not be on the same page. Do we want to allow more than one CheckPoint active or not? I thought we had settled on only one. I can see why this is useful if we go with the design if you allow multiple instances